lib/handlers/router-entities/route-caching/dynamo-route-caching-provider.ts

```diff
@@ -238,39 +296,45 @@ export class DynamoRouteCachingProvider extends IRouteCachingProvider {
       blocksToLive: first.blocksToLive,
     })
 
-    metric.putMetric(`Unique${cachedRoutesMetricName}Found`, cachedRoutes.routes.length, MetricLoggerUnit.Count)
+    metric.putMetric(`Unique${cachedRoutesSource}Found`, cachedRoutes.routes.length, MetricLoggerUnit.Count)
 
     log.info({ cachedRoutes }, `[DynamoRouteCachingProvider] Returning the cached and unmarshalled route.`)
 
     const blocksDifference = currentBlockNumber - blockNumber
-    metric.putMetric(`${cachedRoutesMetricName}BlockDifference`, blocksDifference, MetricLoggerUnit.Count)
+    metric.putMetric(`${cachedRoutesSource}BlockDifference`, blocksDifference, MetricLoggerUnit.Count)
     metric.putMetric(
-      `${cachedRoutesMetricName}BlockDifference_${ID_TO_NETWORK_NAME(chainId)}`,
+      `${cachedRoutesSource}BlockDifference_${ID_TO_NETWORK_NAME(chainId)}`,
       blocksDifference,
       MetricLoggerUnit.Count
     )
 
     const notExpiredCachedRoute = cachedRoutes.notExpired(currentBlockNumber, optimistic)
     if (notExpiredCachedRoute) {
-      metric.putMetric(`${cachedRoutesMetricName}NotExpired`, 1, MetricLoggerUnit.Count)
+      metric.putMetric(`${cachedRoutesSource}NotExpired`, 1, MetricLoggerUnit.Count)
     } else {
-      metric.putMetric(`${cachedRoutesMetricName}Expired`, 1, MetricLoggerUnit.Count)
+      metric.putMetric(`${cachedRoutesSource}Expired`, 1, MetricLoggerUnit.Count)
     }
 
     if (
       optimistic && // If we are in optimistic mode
       cachedRoutes.blockNumber < currentBlockNumber && // and the cachedRoutes are from a block lower than current
-      notExpiredCachedRoute // and the cachedRoutes are not expired
+      notExpiredCachedRoute // and the cachedRoutes are not expired (if they are expired, the regular request will insert cache)
     ) {
-      // We send an async caching quote
-      // we do not await on this function, it's a fire and forget
-      this.maybeSendCachingQuote(partitionKey, sortKey, amount)
+      if (cachedRoutesSource === CachedRoutesSource.CachedRoutes) {
+        // We send an async caching quote
+        // we do not await on this function, it's a fire and forget
+        this.maybeSendCachingQuoteForCachedRoutes(partitionKey, sortKey!, amount)
```

Comment:
how are we sure that this will keep running even after the lambda has returned?

Reply:
we don't know, and that's okay. If this fails we will have more caching requests (so we will monitor it, but not a concern IMO)

Reply:
yeah it will almost def stop once it returns, we'd need a step function for this to work. Because by definition this is slower than the cached response right?

Reply:
well, this is an `Event` type invocation, so it's an async call, the request will response with a 200 to indicate that the lambda received the event but it will be processed asynchronously